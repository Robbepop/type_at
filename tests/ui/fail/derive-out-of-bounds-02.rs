use type_at::TypeAt;

#[derive(TypeAt)]
pub struct Struct { a: i8, b: i16 }

fn main() {
    let _: _  = <Struct as TypeAt<2>>::Type::default();
}
