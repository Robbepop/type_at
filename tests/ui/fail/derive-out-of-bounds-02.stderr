error[E0277]: the trait bound `Struct: TypeAt<2_usize>` is not satisfied
 --> tests/ui/fail/derive-out-of-bounds-02.rs:7:17
  |
7 |     let _: _  = <Struct as TypeAt<2>>::Type::default();
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `TypeAt<2_usize>` is not implemented for `Struct`
  |
  = help: the following implementations were found:
            <Struct as TypeAt<0_usize>>
            <Struct as TypeAt<1_usize>>
